tests/integration/test_ignore.py

```diff
@@ -2,54 +2,73 @@
 import pytest
 from six.moves.urllib.request import urlopen, Request
 from six.moves.urllib.error import HTTPError
+from contextlib import contextmanager
 import vcr
 
+import socket
 
-def test_ignore_localhost(tmpdir, httpserver):
-    httpserver.serve_content('Hello!')
-    cass_file = str(tmpdir.join('filter_qs.yaml'))
-    with vcr.use_cassette(cass_file, ignore_localhost=True) as cass:
-        urlopen(httpserver.url)
-        assert len(cass) == 0
-        urlopen('http://httpbin.org')
-        assert len(cass) == 1
-
-
-def test_ignore_httpbin(tmpdir, httpserver):
-    httpserver.serve_content('Hello!')
-    cass_file = str(tmpdir.join('filter_qs.yaml'))
-    with vcr.use_cassette(
-        cass_file,
-        ignore_hosts=['httpbin.org']
-    ) as cass:
-        urlopen('http://httpbin.org')
-        assert len(cass) == 0
-        urlopen(httpserver.url)
-        assert len(cass) == 1
-
-
-def test_ignore_localhost_and_httpbin(tmpdir, httpserver):
-    httpserver.serve_content('Hello!')
-    cass_file = str(tmpdir.join('filter_qs.yaml'))
-    with vcr.use_cassette(
-        cass_file,
-        ignore_hosts=['httpbin.org'],
-        ignore_localhost=True
-    ) as cass:
-        urlopen('http://httpbin.org')
-        urlopen(httpserver.url)
-        assert len(cass) == 0
-
-def test_ignore_localhost_twice(tmpdir, httpserver):
-    httpserver.serve_content('Hello!')
-    cass_file = str(tmpdir.join('filter_qs.yaml'))
-    with vcr.use_cassette(cass_file, ignore_localhost=True) as cass:
-        urlopen(httpserver.url)
-        assert len(cass) == 0
-        urlopen('http://httpbin.org')
-        assert len(cass) == 1
-    with vcr.use_cassette(cass_file, ignore_localhost=True) as cass:
-        assert len(cass) == 1
-        urlopen(httpserver.url)
-        urlopen('http://httpbin.org')
-        assert len(cass) == 1
+
+@contextmanager
+def overridden_dns(overrides):
+    """
+    Monkeypatch socket.getaddrinfo() to override DNS lookups (name will resolve
+    to address)
+    """
+    real_getaddrinfo = socket.getaddrinfo
+    def fake_getaddrinfo(*args, **kwargs):
+        if args[0] in overrides:
```

Comment:
would aid understanding if you added
`foo, bar = args[0], args[1]`
(or expanded args in the arguments list)